x-pack/plugins/maps/public/classes/fields/agg_field.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IndexPattern } from 'src/plugins/data/public';
+import { IField } from './field';
+import { AGG_TYPE } from '../../../common/constants';
+import { CountAggField } from './count_agg_field';
+import { isMetricCountable } from '../util/is_metric_countable';
+import { IESAggFieldParams } from './agg_field_types';
+import { addFieldToDSL, getField } from '../../../common/elasticsearch_util';
+
+const TERMS_AGG_SHARD_SIZE = 5;
+
+export interface IESFieldedAggParams extends IESAggFieldParams {
+  esDocField?: IField;
+  aggType: AGG_TYPE;
+}
+
+export class AggField extends CountAggField {
+  private readonly _esDocField?: IField;
+  private readonly _aggType: AGG_TYPE;
+
+  constructor(params: IESFieldedAggParams) {
+    super(params);
+    this._esDocField = params.esDocField;
+    this._aggType = params.aggType;
+  }
+
+  _getESDocFieldName(): string {
```

Comment:
This method is only used in `getRootName`. How about just removing the method and making `getRootName` return `this._esDocField ? this._esDocField.getName() : ''`?